server/datastore/mysql/mysql.go

```diff
@@ -146,6 +146,25 @@ func (d *Datastore) withRetryTxx(ctx context.Context, fn txFn) (err error) {
 	return backoff.Retry(operation, bo)
 }
 
+func (d *Datastore) withRetryWriter(fn txFn) (err error) {
```

Comment:
Nit: I can see it's using `d.writer`, but maybe more clear for readers if we rename to something like `withRetryNoTransaction/withRetryNoTx`?